File: JeffSebastian/playground/src/main/java/nl/uva/se/ast/Node.java
package nl.uva.se.ast;

import java.util.Collections;
import java.util.List;

public abstract class Node<T> {

	private List<T> children;

	public Node() {
		children = Collections.<T>emptyList();
	}
	
	public Node(List<T> children) {
		this.children = children;
	}

	public List<T> getChildren() {
		return children;
	}
	
	public boolean hasChildren() {
		return children.size() > 0;
	}

}

File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UvA.SoftCon.Questionnaire.AST.Model
{
    public abstract class Node : INode
    {
        public TextPosition Position
        {
            get;
            private set;
        }

        protected Node(TextPosition position)
        {
            Position = position;
        }

        public abstract void Accept(IASTVisitor visitor);

        public abstract T Accept<T>(IASTVisitor<T> visitor);
    }
}
